deepcell_applications/utils_test.py

```diff
@@ -135,3 +139,77 @@ def test_get_predict_kwargs(mocker):
         bad_namespace = copy.copy(mock_namespace)
         del bad_namespace['test']
         _ = dca.utils.get_predict_kwargs(bad_namespace)
+
+
+def test_get_arg_parser():
+    with tempfile.TemporaryDirectory() as temp_dir:
+        img = np.zeros((10, 10))
+
+        # generate good paths
+        file_path = os.path.join(temp_dir, 'img.tiff')
+        dir_path = os.path.join(temp_dir, 'dir')
+
+        io.imsave(file_path, img)
+        os.makedirs(dir_path)
+
+        # generate bad paths
+        bad_file_path = os.path.join(temp_dir, 'fake_img.tiff')
+        bad_dir_path = os.path.join(temp_dir, 'fake_dir')
+
+        # create dict to hold expected outputs
+        output_dict = {
+            'app': 'mesmer',
+            'output_directory': dir_path,
+            'output_name': 'seg_mask.tif',
+            'log_level': 'INFO',
+            'squeeze': True,
+            'nuclear_path': file_path,
+            'nuclear_channel': [2],
+            'membrane_path': file_path,
+            'membrane_channel': [3],
+            'compartment': 'nuclear',
+            'image_mpp': 3.0,
+            'batch_size': 5}
+
+        # construct syntax for appropriate passing to argparse
+        input_list = [output_dict['app'],
+                      '--output-directory', output_dict['output_directory'],
+                      '--output-name', output_dict['output_name'],
+                      '--log-level', output_dict['log_level'],
+                      '--squeeze',
+                      '--nuclear-image', output_dict['nuclear_path'],
+                      '--nuclear-channel', str(output_dict['nuclear_channel'][0]),
+                      '--membrane-image', output_dict['membrane_path'],
+                      '--membrane-channel', str(output_dict['membrane_channel'][0]),
+                      '--compartment', output_dict['compartment'],
+                      '--image-mpp', str(int(output_dict['image_mpp'])),
+                      '--batch-size', str(output_dict['batch_size'])]
+
+        ARGS = dca.utils.get_arg_parser().parse_args(input_list)
+
+        args_dict = vars(ARGS)
+        # argparser sometimes adds '/private` to the front of the path
+        if 'private' in args_dict['output_directory']:
+            args_dict['output_directory'] = args_dict['output_directory'].split('/private')[1]
```

Comment:
I'm not a fan of this block. Let's learn a bit more about why the output path doesn't seem to always be correct.

If it's always in the `/private` folder then use `.startswith()` instead of `in`, the latter is not specific enough.

I'm